src/io/jpg2proto_encoder.cc

```diff
@@ -16,21 +16,24 @@
  * limitations under the License.
  */
 
-#include "singa/io/image2jpg_encoder.h"
-using namespace cv;
-using namespace std;
+#include "./jpg2proto_encoder.h"
+#include "singa/proto/model.pb.h"
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
 
 namespace singa {
 
 namespace io{
-  string Image2JPGEncoder::Encode(vector<Tensor>& data) {
+  void JPG2ProtoEncoder::Setup(const LayerConf& conf) { return; }
+
+  std::string JPG2ProtoEncoder::Encode(vector<Tensor>& data) {
     // suppose data[0]: data, data[1]: label
     // data[0] has a shape as {height, width, channel}
     CHECK_EQ(data[0].nDim(), 3u);
     int height = data[0].shape()[0];
     int width = data[0].shape()[1];
     int channel = data[0].shape()[2];
-    Mat mat = Mat(height, width, CV_8UC3);
+    cv::Mat mat = cv::Mat(height, width, CV_8UC3, cv::Scalar(0, 0, 0));
```

Comment:
cv2
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
namespace is still cv